File: crates/codegraph-server/tests/fixtures/languages/sample.jl
module MyApp

using DataFrames
import JSON: parse as json_parse

struct User
    name::String
    email::String
end

function create_user(name::String, email::String)::User
    return User(name, email)
end

function greet(user::User)
    if isempty(user.name)
        println("Hello, stranger")
    else
        println("Hello, $(user.name)")
    end
end

end

File: crates/codegraph-server/tests/fixtures/languages/sample.ml
open Printf

type user = {
  name: string;
  email: string;
}

let create_user name email =
  { name; email }

let greet user =
  if user.name = "" then
    printf "Hello, stranger\n"
  else
    printf "Hello, %s\n" user.name
